frontend/src/components/DefenceBox/DefenceConfigurationInput.tsx

```diff
@@ -61,18 +56,16 @@ function DefenceConfigurationInput({
 		}
 	}
 
-	function validateNewInput(value: string) {
-		setConfigValidated(validateDefence(defence.id, config.id, value));
-	}
-
 	const inputElement =
 		inputType === 'text' ? (
 			<ThemedTextArea
 				id={id}
 				content={value}
 				onContentChanged={setValue}
-				configValidated={configValidated}
-				validateInput={validateNewInput}
+				configInvalid={configInvalid}
+				validateInput={(value) => {
+					setConfigInvalid(!validateNewInput(value));
+				}}
```

Comment:
As you have this function call in two places, it might be worth pulling it out into a function, as you originally had. It's also making my head spin a bit with the double negatives!

We might be able to improve the readability (or not!) by using a state var that is "positive" in this DefenceConfigurationInput component, i.e. "configIsValid", and only invert it where you pass it into the Themed components, e.g.

```
const [isConfigValid, setConfigValid] = useState<boolean>(true);

...

function validateConfigValue(value: string) {
  setConfigValid(validateNewInput(value));
}

...

<ThemedTextArea
  valueInvalid={!isConfigValid}
  validateInput={validateConfigValue}
```

Note that the Themed components are by design generic, so use "value" or "input" instead of "config" in prop names.